R/code.R

```diff
@@ -1,224 +1,173 @@
+  if (is.single.string(dataset.modified)) {
+    config$dataset_modified <- if (dataset.modified != "") anytime::iso8601(anytime::anytime(dataset.modified))
+  } else {
+    stop("Warning: 'dataset_modified' must be a single nonempty string.")
   }
-}
-
-assign.column.code.list <- function(config, label, code.list){
-  if (is.boolean(code.list)){
-    eval(str2expression(paste('config$columns$', label, '<- append(', 'config$columns$', label,', list(code_list=', code.list, '\'))', sep='')))
-  } else if(is_valid_url(code.list)){
-    eval(str2expression(paste('config$columns$', label, '<- append(', 'config$columns$', label,', list(code_list=', code.list, '\'))', sep='')))
-  } else if (code.list == ''){
-    return(config)
-  } else{
-    stop('Warning: code.list must be a boolean or a uri.')
+  if (is.valid.uri(license)) {
+    config$license <- license
+  } else if (license != "") {
+    stop("Warning: 'license' must be a URI .")
   }
-}
-
-assign.themes <- function(config, themes){
-  if(length(themes)>1)
-    config$themes = themes
-  else if(themes != '')
-    config$themes = themes
-  else if(themes == '')
-    return(config)
-  else
-    stop('Warning: \'themes\' must be a single string or a vector of strings.')
+  if (is.valid.uri(public.contact.point.uri)) {
+    config$public_contact_point_uri <- public.contact.point.uri
+  } else if (public.contact.point.uri != "") {
+    stop("Warning: 'public.contact.point.uri' must be a URI.")
+  }
+  config$column_names <- names(df)
+  config$columns <- list()
   return(config)
 }
 
-assign.keywords <- function(config, keywords){
-  if(length(keywords)>1)
-    config$keywords = keywords
-  else if(keywords!='')
-    config$keywords = keywords
-  else if(keywords == '')
-    return(config)
-  else
-    stop('Warning: \'keywords\' must be a single string or a vector of strings.')
+#'private function
+#' @noRd
+add.column.configuration <- function(config,
+                                     column.name = "",
+                                     label = "",
+                                     type = "",
+                                     description = "",
+                                     from.existing = "",
+                                     definition.uri = "",
+                                     uri.override = "",
+                                     cell.uri.template = "",
+                                     code.list = "",
+                                     from.template = "",
+                                     data.type = "",
+                                     required = "",
+                                     values = "",
+                                     unit = "",
+                                     measure = "",
+                                     unit.values = "") {
+  if (column.name %in% config$column_names) {
+    eval(str2expression(paste("config$columns<- append(config$columns, list(", column.name, "= list()", "))", sep = "")))
```

Comment:
I thought we could git rid of all of the `eval(str2expression(...))` code now? Could you have a go at removing it all? It'll make the code much easier to read.